qutip/nonmarkov/bofin_baths.py

```diff
@@ -0,0 +1,643 @@
+                new_exponents.append(BathExponent(
+                    exp1.type, None, exp1.Q, ck, exp1.vk, tag=exp1.tag,
+                ))
+            else:
+                # the group includes both type I and R exponents
+                ck_R = (
+                    sum(exp.ck for exp in combine if exp.type == exp.types.R) +
+                    sum(exp.ck for exp in combine if exp.type == exp.types.RI)
+                )
+                ck_I = (
+                    sum(exp.ck for exp in combine if exp.type == exp.types.I) +
+                    sum(exp.ck2 for exp in combine if exp.type == exp.types.RI)
+                )
+                new_exponents.append(BathExponent(
+                    "RI", None, exp1.Q, ck_R, exp1.vk, ck2=ck_I,
+                    tag=exp1.tag,
+                ))
+
+        return new_exponents
+
+
+class DrudeLorentzBath(BosonicBath):
+    """
+    A helper class for constructing a Drude-Lorentz bosonic bath from the
+    bath parameters (see parameters below).
+
+    Parameters
+    ----------
+    Q : Qobj
+        Operator describing the coupling between system and bath.
+
+    lam : float
+        Coupling strength.
+
+    T : float
+        Bath temperature.
+
+    Nk : int
+        Number of exponential terms used to approximate the bath correlation
+        functions.
+
+    gamma : float
+        Bath spectral density cutoff frequency.
+
+    terminator : bool, default False
+        Whether to calculate the Matsubara terminator for the bath. If
+        true, the calculated terminator and the calculated approximation
+        discrepancy are provided via the ``.terminator`` and ``.delta``
+        attributes, respectively. Otherwise, ``.terminator`` and ``.delta`` are
+        both set to ``None``. The terminator is a Liouvillian term, so it's
+        dimensions are those of a superoperator of ``Q``.
+
+    combine : bool, default True
+        Whether to combine exponents with the same frequency (and coupling
+        operator). See :meth:`BosonicBath.combine` for details.
+
+    tag : optional, str, tuple or any other object
+        A label for the bath exponents (for example, the name of the
+        bath). It defaults to None but can be set to help identify which
+        bath an exponent is from.
+
+    Attributes
+    ----------
+    terminator : Qobj
+        The Matsubara terminator -- i.e. a liouvillian term representing the
+        contribution to the system-bath dynamics of all exponential expansion
+        terms beyond Nk. It should be used by adding it to the system
+        liouvillian (i.e. ``liouvillian(H_sys)``).
+
+    delta : float
+        The approximation discrepancy. That is, the difference between the true
+        correlation function of the Drude bath and the correlation function
+        represented by this object is approximately ``2 delta * dirac(t)``,
+        where ``dirac(t)`` denotes the Dirac delta function.
```

Comment:
by ''this object'' i guess you mean the N_k correlation terms included in the heom?  maybe we can also clarify when this approximation is ok